src/Controller/Controller.php

```diff
@@ -693,14 +698,31 @@ public function isAction($action)
     }
 
     /**
+     * {@inheritdoc
+     *
+     * @param string $name Name of the event.
+     * @param array|null $data Any value you wish to be transported with this event to
+     * it can be read by listeners.
+     * @param Controller|null $controller The controller that this event applies to
+     * ($this by default).
+     *
+     * @return ControllerEvent
+     */
+    public function dispatchEvent($name, $data = null, $controller = null)
+    {
+        return $this->_dispatchEvent($name, $data, $controller);
+    }
+
+
+    /**
      * Called before the controller action. You can use this method to configure and customize components
      * or perform logic that needs to happen before each controller action.
      *
-     * @param Event $event An Event instance
+     * @param ControllerEvent $event An Event instance
      * @return void
      * @link http://book.cakephp.org/3.0/en/controllers.html#request-life-cycle-callbacks
      */
-    public function beforeFilter(Event $event)
+    public function beforeFilter(ControllerEvent $event)
```

Comment:
When changing these callbacks back to ```Event``` it should still be possible to use ```ControllerEvent``` without causing any errors in app code, right?